discovery_engine_core/core/src/engine.rs

```diff
@@ -76,6 +81,40 @@ pub struct Config {
     kpe_classifier: String,
 }
 
+#[allow(dead_code)]
+/// Creates a discovery engine with [`xayn_ai::ranker::Ranker`] as a ranker.
+fn create_engine(
+    config: Config,
+    stacks_ops: Vec<BoxedOps>,
+    state: Option<&[u8]>,
+) -> Result<Engine<impl Ranker>, GenericError> {
```

Comment:
since this is a constructor of `Engine`, why don't move this into an impl block?
```suggestion
impl Engine<xayn_ai::ranker::Ranker> {
    /// Creates a discovery engine with [`xayn_ai::ranker::Ranker`] as a ranker.
    #[allow(dead_code)]
    fn create_engine(
        config: Config,
        stacks_ops: Vec<BoxedOps>,
        state: Option<&[u8]>,
    ) -> Result<Self, GenericError> {
```